uniflow/op/extract/split/recursive_character_splitter.py

```diff
@@ -30,12 +33,23 @@ def __init__(
             chunk_size (int): Maximum size of chunks to return.
             chunk_overlap_size (int): Overlap in characters between chunks.
             separators (List[str]): Separators to use.
-            keep_separator: Whether to keep the separator in the chunks.
+            splitting_mode (str): "char" for character count, "token" for token count. Defaults to "char".
         """
         super().__init__(name)
         self._chunk_size = chunk_size
         self._chunk_overlap_size = chunk_overlap_size
         self._separators = separators or self.default_separators
+        self._splitting_mode = splitting_mode  # Track splitting mode
+        if self._splitting_mode == "token":
+            self._encoder = tiktoken.encoding_for_model(
```

Comment:
nit: `self._encoder` is only used once. So the `if` statement could be removed.

Reply:
Done